### PR TITLE
Add solarized palette as named colors

### PR DESCRIPTION
**Pull request by josesho**
## PR Summary
- Adds Ethan Schoonover's 16-color Solarized palette as named colors to matplotlib. They can be accessed by appending 'solarized:' to the relevant color name.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/color/named_colors.py [hunk 1]
```diff
@@ -16,7 +16,8 @@
 from matplotlib import colors as mcolors
 
 
-colors = dict(mcolors.BASE_COLORS, **mcolors.CSS4_COLORS)
+colors = dict(mcolors.BASE_COLORS, **mcolors.CSS4_COLORS,
+                **mcolors.SOLARIZED_COLORS)
```
Comment (timhoffm) [inline examples/color/named_colors.py]: Please align the continued line with the inside I’d the parentheses, I.e. remove two spaces.

File: lib/matplotlib/_color_data.py [hunk 2]
```diff
@@ -1142,3 +1142,26 @@
...
     'yellow':               '#FFFF00',
     'yellowgreen':          '#9ACD32'}
+
+# These are the 16 colors of the Solarized palette by Ethan Schoonover.
+# See https://ethanschoonover.com/solarized/
+# License: https://github.com/altercation/solarized/blob/master/LICENSE
+# Copyright (c) 2011 Ethan Schoonover
+SOLARIZED_COLORS = {
+    'solarized-base03':              '#002b36',
+    'solarized-base02':              '#073642',
```
Comment (timhoffm) [inline lib/matplotlib/_color_data.py]: Please align the color values with the color values of the CSS colors above.

File: lib/matplotlib/_color_data.py [hunk 3]
```diff
@@ -1142,3 +1142,26 @@
     'whitesmoke':           '#F5F5F5',
     'yellow':               '#FFFF00',
     'yellowgreen':          '#9ACD32'}
+
+# These are the 16 colors of the Solarized palette by Ethan Schoonover.
+# See https://ethanschoonover.com/solarized/
+# License: https://github.com/altercation/solarized/blob/master/LICENSE
+# Copyright (c) 2011 Ethan Schoonover
+SOLARIZED_COLORS = {
+    'solarized-base03':              '#002b36',
```
Comment (anntzer) [inline lib/matplotlib/_color_data.py]: The namespacing method we use (cf. xkcd, tab) is "solarized:base03" (colon), not "solarized-base03" (dash).

**Comment by josesho** [comment]
Have addressed your comments; thanks for the review.

**Comment by jklymak** [comment]
This largely seems good to me, but 
https://12757-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/color/named_colors.html#sphx-glr-gallery-color-named-colors-py

doesnt' seem an optimal way to display these.  Also, Tableau colors are not displayed.  Maybe each color should get its own subsection?

**Comment by josesho** [comment]
@jklymak Good idea, I will probably open a new PR to fix this.

**Comment by josesho** [comment]
@jklymak Have updated the named colors example to address your suggestions. Let me know if it's OK!

**Comment by josesho** [comment]
Noted, look our for a new PR here!

**Comment by timhoffm** [comment]
It's really a bit difficult to exactly control sizes and positions in such a plot. Try this for the color table:

~~~
import matplotlib.pyplot as plt
from matplotlib.colors import rgb_to_hsv, to_rgba

def plot_colortable(colors, title, sort_colors=True, ncols=4):
    extra_rows = 2  # additional space for title
    cell_width = 225
    cell_height = 30
    swatch_width = 50
    
    # Sort colors by hue, saturation, value and name.
    by_hsv = ((tuple(rgb_to_hsv(to_rgba(color)[:3])), name)
                    for name, color in colors.items())
    if sort_colors is True:
        by_hsv = sorted(by_hsv)
    names = [name for hsv, name in by_hsv]

    n = len(names)
    nrows = (n + 1) // ncols

    width = cell_width * ncols
    height = cell_height * (nrows + extra_rows)
    dpi = 72
    fig, ax = plt.subplots(figsize=(width / dpi, height / dpi), dpi=dpi)
    
    ax.set_xlim(0, width)
    ax.set_ylim(height, 0)
    ax.yaxis.set_visible(False)
    ax.xaxis.set_visible(False)
    ax.set_axis_off()
    ax.text(0, cell_height, title, fontsize=20)
    
    for i, name in enumerate(names):
        row = i % nrows
        col = i // nrows
        y = (row + extra_rows) * cell_height

        swatch_start_x = cell_width * col
        swatch_end_x = cell_width * col + swatch_width
        text_pos_x = cell_width * col + swatch_width + 5

        ax.text(text_pos_x, y, name, fontsize=12,
                horizontalalignment='left',
                verticalalignment='center')

        ax.hlines(y, swatch_start_x, swatch_end_x,
                  color=colors[name], linewidth=20)
    plt.show()

colortable(mcolors.BASE_COLORS, "Base Colors", sort_colors=False, ncols=3)
colortable(mcolors.CSS4_COLORS, "CSS Colors")
colortable(mcolors.SOLARIZED_COLORS, "Solarized Palette", sort_colors=False)
colortable(mcolors.TABLEAU_COLORS, "Tableau Palette")
~~~

**Comment by josesho** [comment]
Tangent: Adding the Solarized palette led me to discover that [Tableau updated their color palette in 2016](https://www.tableau.com/about/blog/2016/7/colors-upgrade-tableau-10-56782). 

Not sure if it's worth it to update the colors? (Not in this PR, of course.) I myself don't personally use Tableau or its palette, so the benefits are unclear to me.

**Comment by jklymak** [comment]
You've got a few flake 8 errors:  

```./examples/color/named_colors.py:18:1: E303 too many blank lines (3)
./examples/color/named_colors.py:20:1: E302 expected 2 blank lines, found 3
1     E302 expected 2 blank lines, found 3
1     E303 too many blank lines (3)
```

**Comment by timhoffm** [comment]
https://12859-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/color/named_colors.html#sphx-glr-gallery-color-named-colors-py

Your changes lost some of the properties, I intentionally included in the proposed code, because I think they add value.

- **Order of the colors:** Since the color boxes are vertically close together, but horizontally separted by the text label, the natural viewing sequence (assumtion what's close together) is vertical. So the colors should be ordered in colums rather than rows.
- **Grouping:** I don't think that every section has to have four columns. The smaller ones can use less columns if this gives a nicer grouping of the colors, e.g. I would use three columns for the base colors so that bgr cmy and bw are each grouped together.
- The CSS color sections got additional top and bottom margins in your example.

Here's what I expect the table to look like (Note: I've not included solarized because I just did a quick coding of the table and did not check out the code of your PR. But of course solarized should be in there at the end.):

grafik

File: doc/users/whats_new.rst [hunk 4]
```diff
@@ -86,6 +86,13 @@ wrap around, they are a good choice for cyclic data such as phase
 angles, compass directions, or time of day. Like *viridis*, *twilight* is
 perceptually uniform and colorblind friendly.
 
+Add Solarized palette to named matplotlib colors
```
Comment (tacaswell) [inline doc/users/whats_new.rst]: Can you put this in it's own file per https://github.com/matplotlib/matplotlib/blob/master/doc/users/next_whats_new/README.rst ?

**Comment by tacaswell** [comment]
:+1: in principle.

Thanks for also taking on the documentation work!

**Comment by ImportanceOfBeingErnest** [comment]
The list of which options there are to specify colors in matplotlib appears in three places:

 * https://matplotlib.org/devdocs/gallery/color/color_demo.html
* https://matplotlib.org/devdocs/tutorials/colors/colors.html
* https://matplotlib.org/devdocs/api/colors_api.html

Ultimately one would need to update all three tables with the new `solarized:...` colors.  
This can also be a new PR of course, I just wanted to meantion it not to be forgotten.

Additional question: Do the "Accent colors" qualify to create a new listed (discrete) colormap from them, similar to how "tableau" colors are in the "tab10" colormap?

**Comment by timhoffm** [comment]
@josesho still interested in adding solarized colors? The color sorting has been handled separately in #12209. So this PR can be simplified again.

Please note also, that we do not write to `whats_new.rst` directly because it would create a lot of conflicts. You should instead create a separate file for the entry as described in https://github.com/matplotlib/matplotlib/tree/master/doc/users/next_whats_new.

**Comment by josesho** [comment]
I'll have a look at #12209 and modify this PR accordingly. 👌

**Comment by jklymak** [comment]
ping @josesho: Pushing to 3.2, but feel free to update when you get a chance...

**Comment by josesho** [comment]
Been snowed under with work but let me get back to this and see what I need to do!

**Comment by jklymak** [comment]
Sorry I should have been clearer; we need some sort of way to manage these lists of named colors.  Otherwise we will have hundreds of them.

**Comment by anntzer** [comment]
More or less in the same way as for the colormap discussion in some other thread, this could basically be implemented as a third-party package.  In fact a quick googling shows that there's already https://pypi.org/project/pygments-solarized/ which exposes the desired colors in `#rrggbb` format:
```
from pygments_solarized import dark as sol
plt.plot([1, 2], c=sol.MAGENTA)
```
which is not much worse than `"solarized:magenta"`...
Hence I am tempted to just close this.

**Comment by tacaswell** [comment]
@josesho Could you put a note about this in the color docs?

**Pull request closed without merging**